tensorrt_llm/_torch/auto_deploy/transform/library/fused_add_rms_norm.py

```
@@ -43,42 +55,102 @@ def _apply(
+
+            # Erase cast_node *before* replacing add's uses, otherwise
+            # replace_all_uses_with would rewrite cast's input to add_out
+            # which sits after cast in the graph → topological violation.
+            if cast_node is not None:
+                if len(cast_node.users) == 0:
+                    graph.erase_node(cast_node)
+                    erased.add(id(cast_node))
+                else:
+                    # Rare: cast has users besides norm. Redirect them to a
+                    # new cast placed after add_out so the ordering is valid.
+                    with graph.inserting_before(list(cast_node.users)[0]):
```

Ruff (0.15.0)

[warning] Prefer `next(iter(cast_node.users))` over single element slice

Replace with `next(iter(cast_node.users))`

(RUF015)